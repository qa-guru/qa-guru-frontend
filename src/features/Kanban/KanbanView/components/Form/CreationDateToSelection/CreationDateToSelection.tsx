import React from "react";
import { ICreationDateToSelection } from "./CreationDateToSelection.types";
import RHF from "../../../../../../shared/InputRHF";

const CreationDateToSelection: React.FC<ICreationDateToSelection> = ({
  control,
}) => {
  return (
    <RHF.InputDatePicker
      control={control}
      name="creationDateTo"
      label="Выберите дату до"
    />
  );
};

export default CreationDateToSelection;
