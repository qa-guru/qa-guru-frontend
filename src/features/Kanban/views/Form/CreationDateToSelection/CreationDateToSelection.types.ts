import { Control, FieldValues } from "react-hook-form";
import { IFilterKanban } from "../Form.types";

export interface ICreationDateToSelection {
  control: Control<IFilterKanban, any>;
}
