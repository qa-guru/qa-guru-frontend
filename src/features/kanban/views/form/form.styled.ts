import { styled } from "@mui/system";
import { Stack } from "@mui/material";

export const StyledColumnStack = styled(Stack)(({ theme }) => ({
  flexDirection: "column",
  gap: theme.spacing(2),
}));

export const StyledRowStack = styled(Stack)(({ theme }) => ({
  flexDirection: "row",
  gap: theme.spacing(2),
  justifyContent: "center",
}));

export const StyledRow = styled(Stack)(({ theme }) => ({
  flexDirection: "row",
  justifyContent: "space-between",
  alignItems: "center",
  gap: theme.spacing(1),
}));
