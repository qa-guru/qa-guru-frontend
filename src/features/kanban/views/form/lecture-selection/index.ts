export { default } from "./lecture-selection";
