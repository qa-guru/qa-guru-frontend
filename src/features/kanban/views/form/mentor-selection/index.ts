export { default } from "./mentor-selection";
