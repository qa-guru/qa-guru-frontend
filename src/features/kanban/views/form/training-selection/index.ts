export { default } from "./training-selection";
