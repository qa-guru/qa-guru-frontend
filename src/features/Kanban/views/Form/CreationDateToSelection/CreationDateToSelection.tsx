import React from "react";
import { ICreationDateToSelection } from "./CreationDateToSelection.types";
import RHF from "../../../../../shared/components/InputRHF";

const CreationDateToSelection: React.FC<ICreationDateToSelection> = ({
  control,
}) => {
  return (
    <RHF.InputDatePicker
      control={control}
      name="creationDateTo"
      label="Созданные до"
    />
  );
};

export default CreationDateToSelection;
