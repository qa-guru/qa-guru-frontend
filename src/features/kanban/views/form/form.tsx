import { FC, useContext, useEffect, useState } from "react";
import {
  AccordionDetails,
  AccordionSummary,
  Box,
  IconButton,
  Typography,
  useMediaQuery,
} from "@mui/material";
import { useForm } from "react-hook-form";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { useTheme } from "@mui/system";
import RefreshIcon from "@mui/icons-material/Refresh";

import {
  StyledAccordion,
  StyledColumnStack,
  StyledRow,
  StyledRowStack,
} from "./form.styled";
import CreationDateFromSelection from "../creation-date-from-selection";
import CreationDateToSelection from "../creation-date-to-selection";
import { LectureSelection, MentorsSelection } from "../../containers";
import { KanbanFormContext } from "../../context/kanban-form-context";
import TrainingSelection from "../../containers/trainings";

const Form: FC = () => {
  const { control, reset, resetField } = useForm({
    defaultValues: {
      lectures: "",
      trainings: "",
      creationDateFrom: "",
      creationDateTo: "",
      mentors: "",
    },
  });

  const [isAccordionExpanded, setAccordionExpanded] = useState(false);
  const theme = useTheme();
  const isDownMd = useMediaQuery(theme.breakpoints.down("md"));

  const {
    lectureId,
    setTrainingId,
    setLectureId,
    setMentorId,
    setCreationDateFrom,
    setCreationDateTo,
  } = useContext(KanbanFormContext);

  const handleReset = () => {
    setTrainingId(null);
    setLectureId(null);
    setMentorId(null);
    setCreationDateFrom(null);
    setCreationDateTo(null);
    reset();
  };

  const handleAccordionToggle = () => {
    setAccordionExpanded(!isAccordionExpanded);
  };

  useEffect(() => {
    if (!lectureId) {
      resetField("lectures");
    }
  }, [lectureId, resetField]);

  return (
    <form>
      {isDownMd ? (
        <StyledAccordion
          expanded={isAccordionExpanded}
          onChange={handleAccordionToggle}
        >
          <AccordionSummary expandIcon={<ExpandMoreIcon />}>
            <Typography>Фильтр</Typography>
          </AccordionSummary>
          <AccordionDetails>
            <StyledColumnStack>
              <TrainingSelection control={control} />
              <LectureSelection control={control} />
              <MentorsSelection control={control} />
              <StyledRowStack>
                <CreationDateFromSelection control={control} />
                <CreationDateToSelection control={control} />
              </StyledRowStack>
              <StyledRowStack>
                <IconButton onClick={handleReset}>
                  <RefreshIcon color={"primary"} />
                </IconButton>
              </StyledRowStack>
            </StyledColumnStack>
          </AccordionDetails>
        </StyledAccordion>
      ) : (
        <StyledRow>
          <TrainingSelection control={control} />
          <LectureSelection control={control} />
          <MentorsSelection control={control} />
          <CreationDateFromSelection control={control} />
          <CreationDateToSelection control={control} />
          <Box>
            <IconButton onClick={handleReset}>
              <RefreshIcon color={"primary"} fontSize={"medium"} />
            </IconButton>
          </Box>
        </StyledRow>
      )}
    </form>
  );
};

export default Form;
