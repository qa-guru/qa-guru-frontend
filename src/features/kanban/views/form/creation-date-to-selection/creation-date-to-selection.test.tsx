import { MemoryRouter } from "react-router-dom";
import { FormProvider, useForm } from "react-hook-form";
import CreationDateToSelection from "./creation-date-to-selection";
import { render } from "../../../../../test/utils-test";
import { IFilterKanban } from "../form.types";

describe("CreationDateToSelection", () => {
  it("renders correctly", () => {
    function TestComponent() {
      const methods = useForm<IFilterKanban>();
      return (
        <MemoryRouter>
          <FormProvider {...methods}>
            <CreationDateToSelection control={methods.control} />
          </FormProvider>
        </MemoryRouter>
      );
    }

    const { asFragment } = render(<TestComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
