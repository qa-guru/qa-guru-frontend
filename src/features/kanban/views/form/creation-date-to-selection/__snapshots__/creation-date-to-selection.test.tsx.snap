// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreationDateToSelection > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      Созданные до
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        inputmode="text"
        placeholder="MM/DD/YYYY"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Созданные до
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
